File: leet-code/344. Reverse String/344_reverse_string.dart
class Solution {
  void reverseString(List<String> s) {
    int front = 0;
    int back = s.length - 1;

    while (front < back) {
      String temp = s[front];
      s[front] = s[back];
      s[back] = temp;

      front += 1;
      back -= 1;
    }
  }
}

void main() {
  Solution solution = Solution();
  solution.reverseString(["h", "e", "l", "l", "o"]);
}

File: leet-code/344. Reverse String/344_reverse_string.py
class Solution:
    def reverseString(self, s: list[str]) -> None:
        front, back = 0, len(s) - 1

        while front < back:
            s[front], s[back] = s[back], s[front]
            front, back = front + 1, back - 1
